src/julia-syntax.scm

```diff
@@ -881,25 +878,30 @@
                        (call (-/ new) . args)
                        (new-call Tname type-params sparams params
                                  (map (lambda (a) (ctor-body a type-params sparams)) args)
-                                 field-names field-types))
+                                 field-names field-types #t))
```

Comment:
Since `#self#` is used elsewhere, is there any risk this will pass along the wrong variable? E.g. from pattern matching to `self` in the code below?

```
julia> struct G{T}
       x
       G{T}() where T = (x = 1; g() = new(x))
       end

julia> G{Int}
G{Int64}

julia> G{Int}()
(::var"#g#5"{Int64}) (generic function with 1 method)

julia> G{Int}()()
ERROR: TypeError: in new, expected DataType, got a value of type var"#g#5"{Int64}
Stacktrace:
 [1] (::var"#g#5"{Int64})()
   @ Main ./REPL[7]:3
 [2] top-level scope
   @ REPL[10]:1
```